### PR TITLE
New shared mixin css

### PR DESCRIPTION
**Pull request by jyothishnt**
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-774

## Description
Create a new shared mixin for lozenges, roundbuttons, etc

## Deployment URL
We dont have one yet as it is still in storybook and not used in any components.

## Views affected
SS, EV and GB

### CODE REVIEW

File: src/ensembl/src/styles/_mixins.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
+$selectedSpeciesBorderRadius: 20px;
+$selectedSpeciesPadding: 6px 20px;
+$deleteButtonPadding: 8px 20px 8px 28px;
+
+@mixin selected-species-common {
```
Comment (azangru) [inline src/ensembl/src/styles/_mixins.scss:5]: I was thinking more in terms of a mixin that's the same for RoundButton, SelectedSpecies, and whatever it is we'll call this component:

image

If that is the case, the mixin shouldn't be called `selected-species-common`, but rather by the shape that it describes (`lozenge`?).

We probably need to confirm with Andrea that SelectedSpecies and RoundButton have the same shape. Perhaps their height is different, in which case the mixin can accept a `height` parameter.

Comment (jyothishnt) [reply src/ensembl/src/styles/_mixins.scss:5]: yes agreed. I will change that name to lozenge

**Comment by jyothishnt** [comment]
All passed in gitlab. So good to review.

**Comment by azangru** [comment]
Aren't you going to update the RoundButton to use the mixin?

File: src/ensembl/src/styles/_mixins.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  display: inline-block;
+  padding: map-get($lozengeDefaults, "padding");
+  border-radius: $borderRadius;
+  line-height: $lineHeight;
+  user-select: none;
+  margin-bottom: 4px;
+
+  &:not(:last-child) {
+    margin-right: 12px;
+  }
```
Comment (azangru) [inline src/ensembl/src/styles/_mixins.scss:19]: That's not the responsibility of a lozenge

Comment (jyothishnt) [reply src/ensembl/src/styles/_mixins.scss:19]: I agree, it should be removed

File: src/ensembl/src/styles/_mixins.scss [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+);
+
+@mixin lozenge($lineHeight: map-get($lozengeDefaults, "line-height"), $borderRadius: map-get($lozengeDefaults, "border-radius")) {
+  display: inline-block;
+  padding: map-get($lozengeDefaults, "padding");
+  border-radius: $borderRadius;
+  line-height: $lineHeight;
+  user-select: none;
+  margin-bottom: 4px;
+}
```
Comment (azangru) [inline src/ensembl/src/styles/_mixins.scss:16]: Have you tried optional arguments and keyword arguments (https://sass-lang.com/documentation/at-rules/mixin)

E.g. why would we even need a map here? Isn't this simpler?

```
lozenge($lineHeight: 1, $borderRadius: 20px, $padding: 6px 20px) {}
```

Also, from the consumer perspective (`RoundButton.scss`), wouldn't named arguments be better:

```
@include lozenge($borderRadius: 15px);
```

hopefully this should help us avoid passing the same arguments as the default ones.

Comment (jyothishnt) [reply src/ensembl/src/styles/_mixins.scss:16]: yes true. Thanks.

**Review by azangru — APPROVED** [review]

**Pull request merged**